content/sodiq_aderibigbe/creating-queues-using-javascript

```diff
@@ -0,0 +1,154 @@
+---
+Title: "Creating Queues Using JavaScript"
+Description: "Creating a queue and implementing several of it's operations using JavaScript"
+DatePublished: "2022-04-28"
+Categories:
+  - "computer-science"
+Tags:
+  - "Algorithms"
+  - "Arrays"
+  - "ES6"
+  - "Classes"
+CatalogContent:
+  - "learn-javascript"
+  - "course/linear-data-structures"
+---
+
+## Creating Queues Using JavaScript
+
+During conversations about data structures in the programming world, queues are heard quite often.
+A queue is a linear data structure that follows the first-in-first-out pattern i.e removal takes place at the front, and addition takes place at the end. Think of it as the checkout point of a grocery store. Customers who walk there first get attended to first.
```

Comment:
I'd love to hear a little more about linear data structures and what it means to conform to this kind of structure. I think it would also be good to do a more detailed comparison on this grocery store checkout counter anecdote in regard to what the customer represents (a data point?) and how the customer that gets attended to then exits the queue.